Fix bug where shadow on NavigationView pane is missing

chingucoding opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Added missing size and alignment properties for the shadow canvas and added the correct row and column properties.
## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Use the syntax "Closes #1234" or "Fixes #5678" so that GitHub will close the issue once the PR is complete. -->
Fixes #1394. 
## How Has This Been Tested?
<!--- Please describe how you tested your changes. -->
Test manually by starting MUX test app.
## Screenshots (if appropriate):
<!--- If you are making visual changes to a Control or adding a new Control, please include screenshots showing the result. -->
![navigationview-missing-shadow](https://user-images.githubusercontent.com/16122379/66053939-7f19c980-e533-11e9-88ac-50fb51708a45.gif)

kaiguo commented on dev/NavigationView/NavigationView.cpp:
```diff
@@ -3611,9 +3614,16 @@ void NavigationView::UpdatePaneShadow()
+        shadowReceiver.HorizontalAlignment(winrt::HorizontalAlignment::Left);
+
+        // Ensure shadow is as wide as the pane when it is open
+        shadowReceiver.Width(OpenPaneLength() + c_paneElevationTranslationZ);
```
Should it be 2*c_paneElevationTranslationZ since there are negative margins on both left and right?

chingucoding replied:
Good that you thought about that, I totally missed that. However, since the shadow does not exceed outside to the left of the NavigationView (afaik), we don't need that.

Note that in the following screenshot, the NavigationView has a margin of 50 and the red border represents the shadow canvas.
![navviewshadow-bug](https://user-images.githubusercontent.com/16122379/66078628-a76cec80-e562-11e9-8230-c58a5efc4bb2.PNG)

Setting the width of the shadow canvas to a larger value only makes it expand to the right.

kaiguo replied:
I see. Now I think the problem comes from the left negative margin.

![Annotation 2019-10-02 143230](https://user-images.githubusercontent.com/4424330/66083734-cbf4a480-e521-11e9-95db-505cdcc56445.png)

The pane is aligned to the left edge of app window. Seems like adding a left negative margin in this case would decrease the content width instead of taking up more space on the left. I think we should remove the -CompactPaneLength() left margin, then shadow receiver width can just be OpenPaneLength(), top and right margin should do the work to make sure there's enough space for shadow.

![Annotation 2019-10-02 144518](https://user-images.githubusercontent.com/4424330/66084381-5ee20e80-e523-11e9-83a4-05031f4a7daf.png)

chingucoding replied:
Should be fine now with latest commit.

kaiguo commented on dev/NavigationView/NavigationView.cpp:
```diff
@@ -3594,6 +3594,9 @@ void NavigationView::UpdatePaneShadow()
                 contentGrid.SetRowSpan(shadowReceiver, contentGrid.RowDefinitions().Size());
+                contentGrid.SetRow(shadowReceiver, 0);
+                contentGrid.SetColumn(shadowReceiver, 0);
+                contentGrid.SetColumnSpan(shadowReceiver, 2);
```
```suggestion
                contentGrid.SetColumnSpan(shadowReceiver, contentGrid.ColumnDefinitions().Size());
```

kaiguo commented:
/azp run

kaiguo commented:
@chingucoding looks like "NavigationView Regression Test" page is crashing, mind taking a look?

chingucoding commented:
> @chingucoding looks like "NavigationView Regression Test" page is crashing, mind taking a look?

Sure, I will look into this later.

chingucoding commented:
I noticed that we retemplate the NavigationView in the regression test page, and in that retemplate we were not adding the ColumnDefinitions to the content grid holding the shadow canvas. To fix this, I added a check that these are actually defined, so we don't try to insert the shadow canvas into columns that don't exist.

kaiguo commented:
/azp run

kaiguo commented:
@chingucoding I think you need to update visual tree masters since margins are changed. I just added some docs about that here https://github.com/microsoft/microsoft-ui-xaml/pull/1409.

chingucoding commented:
> @chingucoding I think you need to update visual tree masters since margins are changed. I just added some docs about that here #1409.

I updated my branch to be up to date with master and updated the as described in the documentation you posted. Any idea, why it now says, that I have changed over 100 files? Is that because of the merging?

kaiguo commented:
This looks like a bad merge. Reset the commit and try this? https://help.github.com/en/articles/merging-an-upstream-repository-into-your-fork

chingucoding commented:
> This looks like a bad merge. Reset the commit and try this? https://help.github.com/en/articles/merging-an-upstream-repository-into-your-fork

Great idea! Now this PR looks more reasonable for the bug it fixes. 😅

kaiguo commented:
/azp run
